Libplanet/Store/BlockSet.cs

```diff
@@ -49,21 +51,57 @@ public override ICollection<Block<T>> Values
                     throw new KeyNotFoundException();
                 }
 
-                Trace.Assert(block.Hash == key);
+                Trace.Assert(block.Hash.Equals(key));
```

Comment:
Should we prefer `Equals()` instead of `==`?

Reply:
Rider warned me that `HashDigest<SHA256>` hadn't implemented this operator, so I made this change.  AFAIK this operator is implemented through *Uno.CodeGen* though.